Content.Server/StationEvents/GameDirectorSystem.cs

```diff
@@ -0,0 +1,430 @@
+    {
+        // Potentially Complete CurrBeat, which is always scheduler.CurrStory[0]
+        if (scheduler.CurrStory.Count > 0)
+        {
+            var beatName = scheduler.CurrStory[0];
+            var beat = scheduler.StoryBeats[beatName];
+
+            if (scheduler.BeatTime > beat.MaxSecs)
+            {
+                // Done with this beat (it's lasted too long)
+                LogMessage($"StoryBeat {beatName} complete. It's lasted {scheduler.BeatTime} out of a maximum of {beat.MaxSecs} seconds.");
+            }
+            else if (scheduler.BeatTime > beat.MinSecs)
+            {
+                // Determine if we meet the chaos thresholds to exit this beat
+                if (!beat.EndIfAnyWorse.Empty && chaos.AnyWorseThan(beat.EndIfAnyWorse))
+                {
+                    // Done with this beat (chaos exceeded set bad level)
+                    LogMessage($"StoryBeat {beatName} complete. Chaos exceeds {beat.EndIfAnyWorse} (EndIfAnyWorse).");
+                }
+                else if(!beat.EndIfAllBetter.Empty && chaos.AllBetterThan(beat.EndIfAllBetter))
+                {
+                    // Done with this beat (chaos reached set good level)
+                    LogMessage($"StoryBeat {beatName} complete. Chaos better than {beat.EndIfAllBetter} (EndIfAllBetter).");
+                }
+                else
+                {
+                    return beat;
+                }
+            }
+            else
+            {
+                return beat;
+            }
+
+            // If we didn't return by here, we are done with this beat.
+            scheduler.CurrStory.RemoveAt(0);
+        }
+        scheduler.BeatTime = 0.0f;
+
+        // Advance in the current story
+        if (scheduler.CurrStory.Count > 0)
+        {
+            // Return the next beat in the current story.
+            var beatName = scheduler.CurrStory[0];
+            var beat = scheduler.StoryBeats[beatName];
+
+            LogMessage($"New StoryBeat {beatName}: {beat.Description}. Goal is {beat.Goal}");
+            return beat;
+        }
+
+        // Need to find a new story. Pick a random one which meets our needs.
+        var stories = scheduler.Stories.Keys.ToList();
+        _random.Shuffle(stories);
+
+        foreach (var storyName in stories)
+        {
+            var story = scheduler.Stories[storyName];
+            if (story.MinPlayers > count.Players || story.MaxPlayers < count.Players)
+            {
+                continue;
+            }
+
+            // A new story was picked. Copy the full list of beats (for us to pop beats from the front as we proceed)
+            scheduler.CurrStory = story.Beats.ShallowClone();
+            scheduler.CurrStoryName = storyName;
+            SetupEvents(scheduler, count);
+            LogMessage($"New Story {storyName}: {story.Description}. {scheduler.PossibleEvents.Count} events to use.");
```

Comment:
The admins have asked me to offer a lot of feedback to them on what is going on. The red text was requested in discord by admins.

Also, putting this in the admin log at least during playtesting the feature will make it much clearer what tuning is required, why the round went the way it did, etc. See the screenshots at the start of the PR.

Reply:
My reasons
1. Admin logs make it much easier to put round events and gamedirector decisions and their results in context with each other, filtered and in sequence
2. Admins running the round often want direction on what the automation knows about what is going on (it's hard to get a global perspective even as a ghost) and may want to make admeme / dungeon master type decisions in line with what the G.D is doing.

Reply:
Also, I've moved this specific LogMessage to a `_sawmill.Info` call, along with a number more relating to beats ending.

Reply:
> > * Admin logs make it much easier to put round events and gamedirector decisions and their results in context with each other, filtered and in sequence
> 
> There are a crazy amount of admin logs per minute, around 1000 on lizard, and their ability to be filtered outside of by player is poor. If you want to be able to see everything in one place, I think a better solution is to use sawmill.info to log whatever should be affecting gamedirector decisions rather than adding it to admin logs and trying to filter out what's important. There's currently no way to determine or estimate round chaos level from any admin log filter, even reading through admin logs manually it'd be very difficult to determine chaos level.
> 
> > * Admins running the round often want direction on what the automation knows about what is going on (it's hard to get a global perspective even as a ghost) and may want to make admeme / dungeon master type decisions in line with what the Game Director is doing.
> 
> I don't think logs are a good way to do this. Having a command to figure out what the director is trying to do, what it thinks the chaos level is, etc at the moment the command is run seems like a much better solution. Opening logs, and filtering to the director isn't as easy as typing something like `directorinfo`. An even better solution may be to add a button to the admin menu that opens a panel with director info, maybe even director controls

As you can see in the screenshot at the top of the PR, when you configure the logging to show GameDirector messages, they are pretty clear and stand alone. 

One advantage of the GameDirector logging the metrics when it does is that it logs them at the moment it makes a decision. It's at that one moment that the Chaos actually means something - we want to know WHY the director made a decision we need to know the current StoryBeat (goal) and the current Metrics that fed into the optimization.

Of course a console command would probably still be useful, they simply don't replace the logs.

Reply:
I agree with CE, the only time this would be used for is debugging and admin logs are just a tool to use during ahelps.

Also in practice admin logs can sometimes take over 10 seconds to return which seems a lot more cumbersome than something realtime.

Reply:
Fair. All my Admin Log experience is shaped from using it in Dev when there is a silly small amount of data to process. I can see how spammy it is just with little old me mucking around on there. Can't imagine a whole station!